external_search.py

```diff
@@ -17,12 +17,25 @@ class ExternalSearchIndex(object):
     work_document_type = 'work-type'
     __client = None
 
+    CURRENT_ALIAS_SUFFIX = '-current'
+    VERSION_RE = re.compile('-v([0-9]+)$')
+
+    @classmethod
+    def reset(cls):
+        """Resets the __client object to None so a new configuration
+        can be applied during object initialization.
+
+        This method is only intended for use in testing.
+        """
+        cls.__client = None
+
     def __init__(self, url=None, works_index=None):
     
         self.log = logging.getLogger("External search index")
 
         # By default, assume that there is no search index.
         self.works_index = None
+        from_configuration = False
```

Comment:
I'd say if the provided string doesn't exist, the initialization code should create both the index and the alias. Otherwise, the code shouldn't change the alias, but it can create an index if that doesn't exist.

So if you provided works-v2 and nothing exists yet, it would create index works-v2 and alias works-current pointing to it.
If you provided works-v2 and works-current exists but works-v2 doesn't, it would create works-v2 but leave works-current the way it was.

Does that make sense?

I don't think the method to swap over the alias needs to be part of this branch, but you can add it now if you want to.